### PR TITLE
Allow admins to reject uploaded parental consents

### PR DESCRIPTION
**Pull request by shaun-technovation**
This will add basic functionality for an admin to reject an uploaded parental consent form.

Basic overview of changes:
- New route/controller action
- New reject button/action on the datagrid
- New student mailer (to notify them about rejected parental consent)

### CODE REVIEW

File: app/controllers/admin/paper_parental_consents_controller.rb [hunk 1]
```diff
@@ -29,6 +29,17 @@ def approve
 
       redirect_to admin_paper_parental_consents_path,
         success: "You approved the parental consent for #{parental_consent.student_profile_full_name}."
+
```
Comment (viviancan) [inline app/controllers/admin/paper_parental_consents_controller.rb:32]: Missing `end` keyword for the `approve` method

File: config/locales/mailers.en.yml [hunk 2]
```diff
@@ -39,6 +39,10 @@ en:
     welcome_returning_judge:
       subject: "Welcome to Technovation %{season_year}!"
 
+  student_mailer:
+    parental_consent_rejected:
+      subject: Issue with parent/guardian consent form
```
Comment (viviancan) [inline config/locales/mailers.en.yml:44]: Should this subject string be in quotes?

**Review by viviancan — APPROVED** [review]
Tried locally, and with the fixed `end` syntax it works as expected! We should discuss what happens if the paper consent is rejected. On the student side, after my parental consent was rejected, I could upload again. I uploaded a different file, but then in the admin table, it still showed as rejected. Nothing major, but something to work out I think! Maybe if it is rejected, they aren't able to upload through the student dashboard again? 🤔 Great work!

File: app/controllers/admin/paper_parental_consents_controller.rb (hunk 1 above)
Comment (shaun-technovation) [reply app/controllers/admin/paper_parental_consents_controller.rb:32]: Ah, great catch on this! I botched this when I was resolving diff conflicts! 😊 🤦 😂

File: config/locales/mailers.en.yml (hunk 2 above)
Comment (shaun-technovation) [reply config/locales/mailers.en.yml:44]: Should be fine w/o quotes, I'm pretty sure the quotes are needed when there is string interpolation. 👍

**Comment by shaun-technovation** [comment]
> Tried locally, and with the fixed `end` syntax it works as expected! We should discuss what happens if the paper consent is rejected. On the student side, after my parental consent was rejected, I could upload again. I uploaded a different file, but then in the admin table, it still showed as rejected. Nothing major, but something to work out I think! Maybe if it is rejected, they aren't able to upload through the student dashboard again? 🤔 Great work!

Ah, great call-out on rejecting/re-uploading! I think for now I'll just make a quick fix so that any re-upload will reset it to a pending status (so it can be approved or rejected again in the admin).

**Comment by shaun-technovation** [comment]
> > Tried locally, and with the fixed `end` syntax it works as expected! We should discuss what happens if the paper consent is rejected. On the student side, after my parental consent was rejected, I could upload again. I uploaded a different file, but then in the admin table, it still showed as rejected. Nothing major, but something to work out I think! Maybe if it is rejected, they aren't able to upload through the student dashboard again? 🤔 Great work!
> 
> Ah, great call-out on rejecting/re-uploading! I think for now I'll just make a quick fix so that any re-upload will reset it to a pending status (so it can be approved or rejected again in the admin).

Just FYI, I made this update in #4142. 👍

**Pull request merged**